### PR TITLE
[EC-313] Add required infrastructure for Azure Load Test

### PR DESCRIPTION
**Pull request by Krusty93**
### List of changes

Add a resource group
Add a vnet e a subnet
Add an Azure Load Test resource

### Motivation and context

Teams would like to use Azure Load Test to perform some load testing. However, it requires no locks on resources having namespace `Microsoft.Network` in the target resource group. We have plenty of resources with locks in the `io-p-common-rg` resource group, that we don't want/can't remove.

Creating a dedicated resource group/vnet is a good way to solve this issue and keep the production workload independent

### Type of changes

- [X] Add new resources
- [ ] Update configuration to existing resources
- [ ] Remove existing resources

### Env to apply

- [ ] DEV
- [ ] UAT
- [X] PROD

### Does this introduce a change to production resources with possible user impact?

- [ ] Yes, users may be impacted applying this change
- [X] No

### Does this introduce an unwanted change on infrastructure? Check terraform plan execution result

- [ ] Yes
- [X] No

### Other information

---

### How to apply

After PR is approved
1. run deploy pipeline from Azure DevOps [io-platform-iac-projects](https://dev.azure.com/pagopaspa/io-platform-iac-projects/_build?view=folders)
2. select PR branch
3. wait for approval

### CODE REVIEW

File: src/load-test/prod/networking.tf [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+  address_space       = ["10.40.0.0/22"]
+
+  tags = local.tags
+}
+
+resource "azurerm_subnet" "weu_load_test" {
+  name                 = "${local.project}-load-test-snet-01"
+  resource_group_name  = azurerm_resource_group.load_test.name
+  virtual_network_name = azurerm_virtual_network.weu_load_test.name
+  address_prefixes     = ["10.40.0.0/27"]
```
Comment (christian-calabrese) [inline src/load-test/prod/networking.tf:14]: I was wondering whether we could use a larger CIDR for this subnet. Did you oversee any other subnets that may be useful in the future to cover the VNet CIDR?

File: src/load-test/prod/networking.tf [hunk 2]
```diff
@@ -0,0 +1,33 @@
+resource "azurerm_virtual_network" "weu_load_test" {
+  name                = "${local.project}-load-test-vnet-01"
+  resource_group_name = azurerm_resource_group.load_test.name
+  location            = azurerm_resource_group.load_test.location
+  address_space       = ["10.40.0.0/22"]
+
+  tags = local.tags
+}
+
```
Comment (christian-calabrese) [inline src/load-test/prod/networking.tf:9]:
```suggestion

resource "azurerm_private_dns_zone_virtual_network_link" "azurewebsites_private_vnet_common" {
  name = azurerm_virtual_network.weu_load_test.name

  virtual_network_id    = azurerm_virtual_network.weu_load_test.id
  resource_group_name   = data.azurerm_private_dns_zone.privatelink_azurewebsites.resource_group_name
  private_dns_zone_name = data.azurerm_private_dns_zone.privatelink_azurewebsites.name

  registration_enabled = false

  tags = local.tags
}

```
I may be wrong, but DNS resolution of private dns records don't work over VNet peering if the private DNS zone is not linked to both VNets

File: src/load-test/prod/data.tf [hunk 3]
```diff
@@ -0,0 +1,4 @@
+data "azurerm_virtual_network" "weu_common" {
+  name                = "${local.legacy_project}-vnet-common"
+  resource_group_name = "${local.legacy_project}-rg-common"
+}
```
Comment (christian-calabrese) [inline src/load-test/prod/data.tf:4]:
```suggestion
}

data "azurerm_private_dns_zone" "privatelink_azurewebsites" {
  name                = "privatelink.azurewebsites.net"
  resource_group_name = "${local.legacy_project}-rg-common"
}
```

To support suggestion in src/load-test/prod/networking.tf

File: src/load-test/prod/networking.tf (hunk 1 above)
Comment (Krusty93) [reply src/load-test/prod/networking.tf:14]: Only if multiple tests are running at the same time, I can't see any other scenario
I tried to keep it small, but I don't see too much difference among this and a slight bigger subnet

File: src/load-test/prod/networking.tf (hunk 2 above)
Comment (Krusty93) [reply src/load-test/prod/networking.tf:9]: That's correct! I've updated the code

**Review by christian-calabrese — APPROVED** [review]
LGTM!

**Pull request merged**